File: SOLID principles/1-Single-responsibility-principle/src/main/java/solution/UserMapper.java
package solution;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

public class UserMapper {

    public User mapJsonToUser(String userJson) throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        return mapper.readValue(userJson, User.class);
    }
}

File: SOLID principles/1-Single-responsibility-principle/dotNET/dotNET/Solution/UserMapper.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace dotNET.Solution
{
    class UserMapper
    {

        public User mapJsonToUser(String userJson)
        {
            return JsonConvert.DeserializeObject<User>(userJson);
        }
    }
}
